storage/netapp_e/na_eseries_storage_system.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/python
+
+# (c) 2016, NetApp, Inc
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+import json
+import os
+from datetime import datetime as dt, timedelta, time
+from time import sleep
+
+from ansible.module_utils.api import basic_auth_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+from ansible.module_utils.urls import open_url
+
+try:
+    from urllib.error import HTTPError
+except ImportError:
+    from urllib2 import HTTPError
+
+DOCUMENTATION = '''
+module: na_eseries_storage_system
+version_added: "2.2"
+short_description: Add/remove arrays from the Web Services Proxy
+description:
+- Manage the arrays accessible via a NetApp Web Services Proxy for NetApp E-series storage arrays.
+extends_documentation_fragment:
+    - auth_basic
+options:
+  ssid:
+    required: true
+    description:
+    - The ID of the array to manage. This value must be unique for each array.
+  state:
+    required: true
+    description:
+    - Whether the specified array should be configured on the Web Services Proxy or not.
+    choices: ['present', 'absent']
+  controller_addresses:
+    required: true
+    description:
+    - The list addresses for the out-of-band management adapter or the agent host.
+  array_wwn:
+    required: false
+    description:
+    - The WWN of the array to manage. Only necessary if in-band managing multiple arrays on the same agent host.
```

Comment:
The default can be None/Null if you want to express that the entry is not necessarily needed by the underlying library/hardware/service.